### DISCUSSION TITLE
RLS create policy for update specific columns

### DISCUSSION THREAD

**Discussion by saidsalti** [supabase/supabase#10513] [Questions]
Hello all,
I have table tb1  that have columns 
col1 ,col2 ... col7
so now 
col1 to col3 i allow to public user updated , 
but col4 to col7 only authinticated user can updated..
can create policy fro this senario ?
thank you

**Comment by GaryAustin1** [comment]
Not possible with RLS.  
Postgres has column grants you can use to set different roles for different operations.  Basically you remove the table grants then add back grants per column. 

You can also use a before update trigger and function to check the role and not allow update for specific roles by setting new.col = old.col.

**Reply by saidsalti** [reply]
Thank you

 clear
